apps/wakunode2/app.nim

```diff
@@ -329,6 +300,72 @@ proc setupWakuApp*(app: var App): AppResult[void] =
 
   ok()
 
+proc getPorts(listenAddrs: seq[MultiAddress]):
+              AppResult[tuple[tcpPort, websocketPort: Option[Port]]] = 
+  
+  var tcpPort, websocketPort = none(Port)
+  
+  for a in listenAddrs:
+    if a.isWsAddress():
+      if websocketPort.isNone():
+        let wsAddress = initTAddress(a).valueOr:
+          return err(error)
+        websocketPort = some(wsAddress.port)
+    elif tcpPort.isNone():
+      let tcpAddress = initTAddress(a).valueOr:
+        return err(error)
+      tcpPort = some(tcpAddress.port)
+
+  return ok((tcpPort: tcpPort, websocketPort: websocketPort))
+
+proc updateNetConfig(app: var App): AppResult[void] =
+
+  var conf = app.conf
+  let (tcpPort, websocketPort) = getPorts(app.node.switch.peerInfo.listenAddrs).valueOr:
+    return err("Could not retrieve ports " & error)
+    
+  if tcpPort.isSome():
+    conf.tcpPort = tcpPort.get()
+    
+  if websocketPort.isSome():
+    conf.websocketPort = websocketPort.get()
+    
+  # Rebuild NetConfig with bound port values
+  let netConf = networkConfiguration(conf, clientId).valueOr:
+    return err("Could not update NetConfig: " & error)
+  
+  app.netConf = netConf
+
+  return ok()
+  
+proc updateEnr(app: var App): AppResult[void] =
+
+  let record = enrConfiguration(app.conf, app.netConf, app.key).valueOr:
+    return err(error)
+
+  app.record = record
+  app.node.enr = record
+
+  if app.conf.discv5Discovery:
+    app.wakuDiscV5 = some(app.setupDiscoveryV5())
+
+  ok()
+
+proc updateApp(app: var App): AppResult[void] =
+  
+  if app.conf.tcpPort == Port(0) or app.conf.websocketPort == Port(0):
+    
+    updateNetConfig(app).isOkOr:
+      return err(error)
+    
+    updateEnr(app).isOkOr:
+      return err(error)
+
+    app.node.announcedAddresses = app.netConf.announcedAddresses
+
+    printNodeNetworkInfo(app.node)
+
+  ok()
```

Comment:
A small nitpick and adding a bit more feedback in case of error:

```suggestion
    updateNetConfig(app).isOkOr:
      return err("error calling updateNetConfig: " & &error)
    
    updateEnr(app).isOkOr:
      return err("error calling updateEnr: " & $error)

    app.node.announcedAddresses = app.netConf.announcedAddresses

    printNodeNetworkInfo(app.node)

  return ok()
```